extensions/ql-vscode/src/queryserver-client.ts

```diff
@@ -86,6 +89,26 @@ export class QueryServerClient extends DisposableObject {
     this.evaluationResultCallbacks = {};
   }
 
+  async initLogger() {
+    let storagePath = this.opts.contextStoragePath;
+    let isCustomLogDirectory = false;
+    if (this.config.customLogDirectory) {
+      try {
+        if (!fs.pathExists(this.config.customLogDirectory)) {
+          await fs.mkdir(this.config.customLogDirectory);
+        }
```

Comment:
Just realised that this logic doesn't quite work 🙃 (When I enter an invalid directory, it doesn't still doesn't `catch` any errors.)

Reply:
Latest commit (https://github.com/github/vscode-codeql/pull/863/commits/8d711c668e5781ec91dfa96f056c7d91e68a8b17) fixes the logic, but with slightly annoying duplication... I haven't found an easy way to tidy this up yet 😞

Reply:
I think `if (!fs.pathExists(this.config.customLogDirectory)) {` should be `if (!(await fs.pathExists(this.config.customLogDirectory))) {` Sorry, I didn't catch that earlier.

Reply:
Would that fix the problem here?

Reply:
That seems to work 🎉  Thank you!